codegen/templates/http_client.tmpl

```diff
@@ -170,35 +177,39 @@ func (c *{{$clientName}}) {{title .Name}}(
 	switch res.StatusCode {
 		case {{.OKStatusCode.Code}}:
 			var responseBody {{unref .ResponseType}}
+			{{if isPointerType .ResponseType -}}
 			err = res.ReadAndUnmarshalBody(&responseBody)
+			{{else -}}
+			err = res.ReadAndUnmarshalNonPointerValue(&responseBody)
+			{{end -}}
 			if err != nil {
-				return nil, respHeaders, err
+				return defaultRes, respHeaders, err
 			}
 
 			{{- if .ResHeaderFields }}
 			// TODO(jakev): read response headers and put them in body
 			{{- end}}
 
-			return &responseBody, respHeaders, nil
+			return {{if isPointerType .ResponseType}}&{{end}}responseBody, respHeaders, nil
```

Comment:
Is it better to use ` {{if isPointerType .ResponseType}}&{{end}}` or should we just modify the response type with an `&` prefix? Haven't looked through all the code, but seems like in a few of these cases that would be cleaner.